source/log-reconciler.ts

```diff
@@ -64,14 +61,16 @@ const ensureOrder = <TLog extends Log>(headLog: TLog | undefined, newLog: TLog)
 export const reconcileLogHistoryWithRemovedBlock = async <TBlock extends Block, TLog extends Log>(
 	logHistory: LogHistory<TLog>|Promise<LogHistory<TLog>>,
 	removedBlock: TBlock,
-	onLogRemoved: (log: TLog) => Promise<void>,
+	onLogsRemoved: (blockHash: string, logs: Array<TLog>) => Promise<void>,
 ): Promise<LogHistory<TLog>> => {
 	logHistory = await logHistory;
 
+	const removedLogs = []
 	while (!logHistory.isEmpty() && logHistory.last().blockHash === removedBlock.hash) {
-		await onLogRemoved(logHistory.last());
+		removedLogs.push(logHistory.last())
```

Comment:
There is mixed semicolon/non-semicolon in these files, is that intentional?

Reply:
Fixed.